# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ahmedmohamed01095032905-code/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [14]:
!git clone https://github.com/ahmedmohamed01095032905-code/flyrank-ml-internship.git
%cd flyrank-ml-internship

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 125, done.
remote: Counting objects: 100% (125/125), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 125 (delta 39), reused 94 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (125/125), 1.85 MiB | 8.47 MiB/s, done.
Resolving deltas: 100% (39/39), done.
/content/flyrank-ml-internship


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I chose **Lane 2: Refresh / Content Opportunity Scoring**.

I chose this lane because it focuses on identifying which pages should be reviewed first based on observable search and content signals. It supports real business decisions by helping prioritize limited review resources using data-driven evidence.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*


## Decision
Which content pages should be reviewed first for refresh or improvement?

## Action
Review and update the highest-ranked pages based on the generated opportunity score.

## Cost of a Wrong Recommendation
If the model recommends the wrong pages, reviewers may waste time while important pages continue losing traffic and visibility.

## Why ML?
Machine learning can combine multiple observable signals better than fixed hand-written rules, helping prioritize pages more accurately.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [17]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(),3))
print("Average Position:", round(df["avg_position"].mean(),2))
print("Declining Pages:", (df["trend_direction"]=="down").sum())

Rows: 30000
Columns: 44
Average CTR: 0.511
Average Position: 16.34
Declining Pages: 16262


## Quick Look at the Data

The starter dataset contains **30,000 content pages** and **44 features** describing search and content performance.

The average CTR is **0.511**, while the average search position is **16.34**.

There are **16,262 pages** labeled as declining, indicating that many pages may require review. These numbers suggest that the dataset contains enough variation to study content refresh opportunities and build a ranking model.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This analysis is observational. The results can identify patterns and help prioritize content review, but they do not prove that refreshing a page will improve its performance or explain how search engine algorithms work.

## Self Check

- I selected a predefined lane (Lane 2).
- I defined the decision, action, and cost of a wrong recommendation.
- I included real numbers from the starter dataset.
- I used careful observational language.
- I understand that this project supports human decision-making rather than replacing it.